src/validation.cpp

```diff
@@ -5298,6 +5298,256 @@ static bool ExpectedAssumeutxo(int height, uint256& expected_out)
     return true;
 }
 
+bool ChainstateManager::ActivateSnapshot(
+        CAutoFile* coins_file,
+        SnapshotMetadata metadata,
+        bool in_memory)
+{
+    uint256 base_blockhash = metadata.m_base_blockhash;
+
+    // Can't activate a snapshot more than once.
+    assert(!this->SnapshotBlockhash());
+    int64_t current_coinsdb_cache_size{0};
+    int64_t current_coinstip_cache_size{0};
+
+    // Cache percentages to allocate to each chainstate.
+    constexpr float ibd_cache_perc = 0.01;
+    constexpr float snapshot_cache_perc = 0.99;
+
+    {
+        LOCK(::cs_main);
+        // Resize the coins caches to ensure we're not exceeding memory limits.
+        //
+        // Allocate the majority of the cache to the incoming snapshot chainstate, since
+        // (optimistically) getting to its tip will be the top priority. We'll need to call
+        // `MaybeRebalanceCaches()` once we're done with this function to ensure
+        // the right allocation (including the possibility that no snapshot was activated
+        // and that we should restore the active chainstate caches to their original size).
+        //
+        current_coinsdb_cache_size = ::ChainstateActive().m_coinsdb_cache_size_bytes;
+        current_coinstip_cache_size = ::ChainstateActive().m_coinstip_cache_size_bytes;
```

Comment:
nit: Please prefer the identical ChainstateManager::ActiveChainstate

Reply:
Oof, dumb move on my part. Thanks, fixed.